src/event_loop.rs

```diff
@@ -93,11 +93,14 @@ impl<T> EventLoopBuilder<T> {
     ///
     /// - **iOS:** Can only be called on the main thread.
     #[inline]
-    pub fn build(&mut self) -> EventLoop<T> {
-        EventLoop {
-            event_loop: platform_impl::EventLoop::new(&self.platform_specific),
+    pub fn build(&mut self) -> Result<EventLoop<T>, String> {
```

Comment:
`String` is not a proper error type; please create a newtype that implements `Display` and `Error` - or perhaps a (non-exhaustive?) `enum` to allow handling the different error cases

Reply:
I added an error enum